### ISSUE TITLE
4.19 Releng

### ISSUE THREAD

**Issue by wimjongman** [eclipse-birt/birt#1997]
Tracks the release engineering activities for the 4.19 release.

### Opening actions
- [x] Update version numbers
- [x] Create Milestone 4.20 in Github
- [x] Move 4.18 open issues to 4.19
- [x] Create Release in [Project Administration](https://projects.eclipse.org/projects/technology.birt)

### Release day - 7
- [x] Announce new release pending in discussions
- [x] Revisit all open tasks

### Release candidate (Milestone) - testing
- [x] BIRT 4.19 All-In-One designer
- [x] BIRT 4.19 runtime
- [x] BIRT 4.19 runtime OSGi

### Release day
- [x] Create Github release (Creates Tag and Announcement in Discussions)
- [x] Update [README.MD](https://github.com/eclipse-birt/birt/blob/master/README.md)
- [x] Update 4.19 Release in [Project Administration](https://projects.eclipse.org/projects/technology.birt)
- [x] Create 4.20 Release in [Project Administration](https://projects.eclipse.org/projects/technology.birt)
- [x] Move 4.19 open issues to 4.20

**Comment by merks** [comment]
FYI, the first nightly build for 4.19 is available:

https://download.eclipse.org/birt/updates/nightly/latest

This includes updates to the set and the target platform to follow latest updates of all dependencies, so be sure to do a pull and Perform Setup Tasks.

**Comment by speckyspooky** [comment]
@wimjongman, @merks, @hvbtup 

Yesterday the issue #2010 was reported for BIRT 4.18 for thedocx-output.
Of course I was able after deep investigation that there is a topic with grin-in-grid/table-in-table and combinations.
The reason is the feature #1957 and the partial removed "empty paragraphs of word".

I provided a fix for that PR #2011.

Question to your side, could we provide a BugFix release of BIRT before original planned March 2025?
Yes, we have 2 options to avoid the error at BIRT 4.18 designer-property & user-property
but at the end it is a very significant bug for docx-output.

So I would interested to hear your points for kind of BugFix-release or release like planned.

**Comment by merks** [comment]
There a difference between could we and should we.  Of course everything is possible given enough resource for the work.  But consumers are getting all this for free and yet they mostly contribute pretty much nothing, including not bothering to test milestones and release candidates.  With a tiny bit of participation folks could prevent problems that affect them being in a release.  The nightly build with a fix is available to anyone.  We can build a milestone version with stable repository URL that will exist until 4.20 development starts after the 4.19 release (and can produce that with just the push of a button) .  Also folks can continue to use 4.17.   Given the free nature of the supply chain, consumers should not expect more, especially not without doing more themselves.

**Comment by wimjongman** [comment]
I agree. People can use 4.17 or 4.19 nightly when 4.18 does not work for them. Besides, who is to say that the user does not find something else. 

Thanks for your thoughts, guys. Thomas, we will take your patch and let the user review it.

**Comment by speckyspooky** [comment]
Thanks for your comments, I'm fine with it and yes, the user have multiple options to change the behavior or use another improved version.

**Comment by wimjongman** [comment]
Thanks for the heads-up.

**Comment by wimjongman** [comment]
Thanks, Ed. I started the closing ceremony and also tested all products.

+1 from me

- [x] BIRT 4.19 All-In-One designer
- [x] BIRT 4.19 runtime
- [x] BIRT 4.19 runtime OSGi

**Comment by merks** [comment]
Given the feedback above, I plan to promote the milestone to a release on Wednesday as part of my other SimRel activities.  

I also plan to make the necessary changes to start the 4.20 cycle.

@wimjongman Perhaps you would like to create a 4.20 Releng issue before Wednesday to track the 4.20 activities...

**Comment by wimjongman** [comment]
https://github.com/eclipse-birt/birt/issues/2080

**Comment by merks** [comment]
The release is published:

https://download.eclipse.org/birt/updates/release/4.19.0

And the repository is tagged:

https://github.com/eclipse-birt/birt/releases/tag/4.19

I think there are a few final steps that @wimjongman usually does.  

I hope to get 4.20 started later today or early tomorrow.

**Comment by wimjongman** [comment]
Thanks, everybody, for this great release!
